src/main/java/net/neoforged/neoforge/common/extensions/IServerCommonPacketListenerExtension.java

```diff
@@ -106,4 +107,9 @@ default boolean isConnected(final ResourceLocation payloadId) {
     default boolean isConnected(final CustomPacketPayload payload) {
         return isConnected(payload.id());
     }
+
+    /**
+     * {@return the connection type of the connection}
+     */
+    ConnectionType getConnectionType();
```

Comment:
I also don't like that `isVanillaConnection` gets its information from a different source than `getConnectionType`.